src/openms/source/KERNEL/MSSpectrum.cpp

```diff
@@ -725,6 +725,19 @@ namespace OpenMS
     return getIonMobilityArray__(this->getFloatDataArrays(), index, unit);
   }
 
+  std::pair<DriftTimeUnit, std::vector<float>> MSSpectrum::maybeGetIMData() const
```

Comment:
maybe also consider FAIMS IM annotation?

Reply:
I think you should discuss with Josh or the open swath people. They put this there.
I have no idea how this would be annotated.

Reply:
I just made it faster by having less if cases and meta lookups

Reply:
In general, I think my PR does not do much with IM apart from this small improvement.
I have not planned anything with aggregating IM dimensions, yet.
You would need to sort back and forth in different dimensions or have a separate data structure to do this efficiently.

Reply:
yeah I added this as a comment to this discussion. I think it is worth thinking a bit about how to handle IM data or if we need a different interface